File: spec/java/src/io/kaitai/struct/spec/TestStrPadTerm.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.StrPadTerm;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestStrPadTerm extends CommonSpec {
    @Test
    public void testStrPadTerm() throws Exception {
        StrPadTerm r = StrPadTerm.fromFile(SRC_DIR + "str_pad_term.bin");

        assertEquals(r.strPad(), "str1");
        assertEquals(r.strTerm(), "str2foo");
        assertEquals(r.strTermAndPad(), "str+++3bar+++");
        assertEquals(r.strTermInclude(), "str4baz@");
    }
}

File: spec/rust/tests/test_str_pad_term.rs
#![allow(dead_code)]
use std::fs;

extern crate kaitai;
use self::kaitai::*;
mod formats;
use formats::str_pad_term::*;

#[test]
fn test_str_pad_term() {
    let bytes = fs::read("../../src/str_pad_term.bin").unwrap();
    let reader = BytesReader::new(&bytes);
    let mut r = StrPadTerm::default();

    if let Err(err) = r.read(&reader, None, Some(KStructUnit::parent_stack())) {

        panic!("{:?}", err);
    }
    assert_eq!(r.str_pad(), "str1");
    assert_eq!(r.str_term(), "str2foo");
    assert_eq!(r.str_term_and_pad(), "str+++3bar+++");
    assert_eq!(r.str_term_include(), "str4baz@");
}
